File: easy/1636_sort_array_by_increasing_frequency/solution.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
  std::vector<int> frequencySort(std::vector<int> &nums) {
    std::unordered_map<int, int> numsCounter;

    for (const int &num : nums)
      ++numsCounter[num];

    sort(nums.begin(), nums.end(), [&](const int &a, const int &b) {
      if (numsCounter[a] != numsCounter[b])
        return numsCounter[a] < numsCounter[b];

      return a > b;
    });

    return nums;
  }
};

File: easy/1636_sort_array_by_increasing_frequency/solution.py
from collections import Counter
from typing import List


class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        sorted_nums = []

        for num, count in sorted(
            Counter(nums).items(), key=lambda t: (t[1], -t[0])
        ):
            sorted_nums.extend([num] * count)

        return sorted_nums
